File: petgo_app/lib/features/consult/domain/consult_history_item.dart
/// 问诊历史条目（对应后端 `ConsultHistoryItem`，Story 5.8）。AI / 兽医两类。
class ConsultHistoryItem {
  const ConsultHistoryItem({
    required this.type,
    required this.date,
    this.triageId,
    this.dangerLevel,
    this.symptomSummary,
    this.sessionId,
    this.vetDisplayName,
    this.sessionSummary,
    this.userStars,
    this.archived,
    this.terminalState,
  });

  final String type; // AI | VET
  final DateTime? date;
  // AI
  final int? triageId;
  final String? dangerLevel;
  final String? symptomSummary;
  // VET
  final int? sessionId;
  final String? vetDisplayName;
  final String? sessionSummary;
  final int? userStars;
  final bool? archived;
  final String? terminalState; // CLOSED | INTERRUPTED

  bool get isAi => type == 'AI';

  factory ConsultHistoryItem.fromJson(Map<String, dynamic> json) => ConsultHistoryItem(
        type: (json['type'] ?? 'AI') as String,
        date: json['date'] == null ? null : DateTime.tryParse(json['date'] as String),
        triageId: (json['triageId'] as num?)?.toInt(),
        dangerLevel: json['dangerLevel'] as String?,
        symptomSummary: json['symptomSummary'] as String?,
        sessionId: (json['sessionId'] as num?)?.toInt(),
        vetDisplayName: json['vetDisplayName'] as String?,
        sessionSummary: json['sessionSummary'] as String?,
        userStars: (json['userStars'] as num?)?.toInt(),
        archived: json['archived'] as bool?,
        terminalState: json['terminalState'] as String?,
      );
}

/// 问诊历史分页（对应后端 `ConsultHistoryPage`）。
class ConsultHistoryPage {
  const ConsultHistoryPage({required this.items, this.nextCursor, required this.hasMore});

  final List<ConsultHistoryItem> items;
  final String? nextCursor;
  final bool hasMore;

  factory ConsultHistoryPage.fromJson(Map<String, dynamic> json) => ConsultHistoryPage(
        items: (json['items'] as List? ?? [])
            .map((e) => ConsultHistoryItem.fromJson((e as Map).cast<String, dynamic>()))
            .toList(),
        nextCursor: json['nextCursor'] as String?,
        hasMore: (json['hasMore'] ?? false) as bool,
      );
}

File: petgo-backend/src/main/java/com/tailtopia/consult/dto/ConsultHistoryItem.java
package com.tailtopia.consult.dto;

import java.time.Instant;

/**
 * 问诊历史统一条目（Story 5.8）。{@code type=AI|VET}，按 {@code date} 倒序混排。
 *
 * <p>AI 项：{@code dangerLevel}+{@code symptomSummary}+{@code triageId}。
 * VET 项：{@code vetDisplayName}+{@code sessionSummary}+{@code userStars}(null=未评分)+{@code archived}
 * +{@code terminalState}(CLOSED/INTERRUPTED)+{@code closedReason}/{@code interruptedReason}+{@code sessionId}。
 * 历史保留<b>独立于存档</b>（{@code archived} 仅标记位）。Jackson NON_NULL 省略空字段。
 */
public record ConsultHistoryItem(
        String type,
        Instant date,
        // AI
        Long triageId,
        String dangerLevel,
        String symptomSummary,
        // VET
        Long sessionId,
        String vetDisplayName,
        String sessionSummary,
        Integer userStars,
        Boolean archived,
        String terminalState,
        String closedReason,
        String interruptedReason) {

    public static ConsultHistoryItem ai(long triageId, String dangerLevel, String symptomSummary, Instant date) {
        return new ConsultHistoryItem("AI", date, triageId, dangerLevel, symptomSummary,
                null, null, null, null, null, null, null, null);
    }

    public static ConsultHistoryItem vet(long sessionId, String vetDisplayName, String sessionSummary,
            Integer userStars, boolean archived, String terminalState, String closedReason,
            String interruptedReason, Instant date) {
        return new ConsultHistoryItem("VET", date, null, null, null,
                sessionId, vetDisplayName, sessionSummary, userStars, archived,
                terminalState, closedReason, interruptedReason);
    }
}
